lib/pure/os.nim

```diff
@@ -1686,6 +1686,26 @@ proc setFilePermissions*(filename: string, permissions: set[FilePermission]) {.
       var res2 = setFileAttributesA(filename, res)
     if res2 == - 1'i32: raiseOSError(osLastError(), $(filename, permissions))
 
+func toFilePerm*(perm: Natural): set[FilePermission] {.since: (1, 3).} =
+  ## Convenience func to convert Unix like file permission to ``set[FilePermission]``.
+  ##
+  ## See also:
+  ## * `getFilePermissions <#getFilePermissions,string>`_
```

Comment:
doesn't have to be in this PR but if we're having this, it'd make sense to have the reverse direction:
```nim
doAssert 0o700.toFilePerm.fromFilePerm == 0o700
```

eg usage:
```nim
from strutils import toOct
doAssert "foo.txt".getFilePermissions.fromFilePerm.toOct == "700"
```

what do others think?

Reply:
`octal.fromMode(): set` and `set.toMode(): octal` would be nice